data_structures/red_black_tree.ts

```diff
@@ -219,7 +217,7 @@ export class RedBlackTree<T> extends BinarySearchTree<T> {
     let unmappedValues: ArrayLike<T> | Iterable<T> = [];
     if (collection instanceof RedBlackTree) {
       result = new RedBlackTree(
-        options?.compare ?? (collection as unknown as RedBlackTree<U>).compare,
```

Comment:
This doesn't work? You can not reference private fields in the superclass from a subclass. That would defeat the whole point of private fields.

The static block thing I did is the proper way to "leak" private field or methods for trusted consumers.

Reply:
Another mistake. Fixed.